src/index.ts

```diff
@@ -220,7 +223,24 @@ async function addResolverOutput(
 	);
 
 	for (const file of extraFiles) {
-		pack.entry({ name: file.name, size: file.size }, file.contents);
+		let content: Buffer | string = file.contents;
+		if (
+			(specifiedDockerfilePath &&
+				file.name === resolver.getCanonicalName(specifiedDockerfilePath)) ||
+			file.name === 'Dockerfile'
```

Comment:
This function in `index.ts` was supposed to be "resolver agnostic", simply calling `resolver.resolve` and letting each resolver decide on what entries to return. This new `if` statement brings resolver-specific logic outside resolvers -- e.g. it does not apply to `nodeResolver`. In another PR comment, I am suggesting that the implementation of `resolver.resolve()` could be modified to include the original entry if the resolver wants it, and that might be a good place to call the dockerfile hook, instead of calling it here. If code duplication becomes an issue, a common function could be extracted, or even an intermediate parent class in a hierarchy, for example:
```
Resolver -> NodeResolver
Resolver -> DockerfileResolver -> PlainDockefileResolver
Resolver -> DockerfileResolver -> ArchDockerfileResolver
Resolver -> DockerfileResolver -> DockerfileTemplateResolver
```
But it may not be needed, the code duplication might be just a few lines and ok to live with.